api/v1alpha1/releaselink_webhook_test.go

```diff
@@ -0,0 +1,164 @@
+//
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	//+kubebuilder:scaffold:imports
+)
+
+const (
+	timeout  = time.Second * 10
+	interval = time.Millisecond * 250
+)
+
+var _ = Describe("ReleaseLink validation webhook", func() {
+
+	// Define utility constants for object names.
+	const (
+		Name            = "test-releaselink-123"
+		Namespace       = "default"
+		DisplayName     = "example-release-link"
+		Application     = "test-application"
+		Target          = "test-target"
+		ReleaseStrategy = "test-releasestrategy"
+	)
+
+	Context("Create ReleaseLink CR with bad fields", func() {
+		It("Should reject until all the fields are valid", func() {
+			ctx := context.Background()
+
+			uniqueName := Name + "1"
```

Comment:
You could replace this with `GenerateName: Name` in line 61 and `Name = "test-releaselink-" in line 40. You can get the resource name once it's created.